homeassistant/components/elkm1/__init__.py

```diff
@@ -301,18 +305,19 @@ def _element_changed(element, changeset):
     return True
 
 
-def _included(ranges, set_to, values):
+def _included(ranges: list[tuple[int, int]], set_to: bool, values: list[bool]) -> None:
     for rng in ranges:
         if not rng[0] <= rng[1] <= len(values):
             raise vol.Invalid(f"Invalid range {rng}")
         values[rng[0] - 1 : rng[1]] = [set_to] * (rng[1] - rng[0] + 1)
 
 
-def _find_elk_by_prefix(hass, prefix):
+def _find_elk_by_prefix(hass: HomeAssistant, prefix: str) -> Elk | None:
     """Search all config entries for a given prefix."""
     for entry_id in hass.data[DOMAIN]:
         if hass.data[DOMAIN][entry_id]["prefix"] == prefix:
-            return hass.data[DOMAIN][entry_id]["elk"]
+            return cast(Elk, hass.data[DOMAIN][entry_id]["elk"])
```

Comment:
In the future we can make `hass.data[DOMAIN][entry_id]` a `dataclass` to avoid having to cast.

Example: https://github.com/home-assistant/core/blob/dev/homeassistant/components/lookin/__init__.py#L155

Reply:
I'd be glad to take this on as a future improvement.